pkg/controllers/user/helm/controller.go

```diff
@@ -396,3 +404,15 @@ func isSame(obj *v3.App, revision *v3.AppRevision) bool {
 
 	return false
 }
+
+func (l *Lifecycle) getHelmVersion(obj *v3.App) (string, error) {
+	templateVersionID, templateVersionNamespace, err := hCommon.ParseExternalID(obj.Spec.ExternalID)
+	if err != nil {
+		return "", err
+	}
+	templateVersion, err := l.TemplateVersionClient.GetNamespaced(templateVersionNamespace, templateVersionID, metav1.GetOptions{})
```

Comment:
This is using a client which is an API call, update to use a lister which will probably require adding one to the lifecycle